source/mesh_deformation/fastscapecc.cc

```diff
@@ -0,0 +1,1064 @@
+
+          // 1. Create the FastScape grid adapter
+// TODO: I haven't looked at the GridAdapter yet, but we should see whether we can't get rid of the const_cast.
+          auto grid = GridAdapterType(const_cast<SurfaceMeshType &>(surface_mesh), surface_cell_area);
+          auto shape = grid.shape();  // Should be {n_grid_nodes}
+
+          // 2. Define the node status array with correct shape and type
+          xt::xtensor<fastscapelib::node_status, 1> node_status_array(shape);
+          std::fill(node_status_array.begin(), node_status_array.end(), fastscapelib::node_status::core);
+          grid.set_nodes_status(node_status_array);
+
+          // // 3. Create the flow graph
+          // auto flow_graph = FlowGraphType(
+          //   grid,
+          //   { fastscapelib::single_flow_router(), fastscapelib::MSTSinkResolver()}
+          // );
+
+          auto flow_graph = FlowGraphType(
+              grid,
+              {
+                  std::make_shared<fastscapelib::single_flow_router>(),
+                  std::make_shared<fastscapelib::mst_sink_resolver>()
+              }
+);
+
+          // 4. Set base level nodes
+          std::vector<std::size_t> base_level_nodes = {0};
+          flow_graph.set_base_levels(base_level_nodes);
+
+          // 5. Build the erosion model
+          fastscapelib::spl_eroder<FlowGraphType> spl_eroder
+          =
+            // fastscapelib::make_spl_eroder(flow_graph, kff, n, m, 1e-5)
+            fastscapelib::make_spl_eroder(flow_graph, 5e-4, 0.4, 1, 1e-5);
+          std::cout << "KFF : " << kff<< std::endl;
+
+          //Only for raster grid 
+          //To propose if we add back the raster grid option later 
+          // auto diffusion_eroder = fastscapelib::make_diffusion_adi_eroder(grid, kdd);
+
+         // uplift rate in meters per year for Fastscape
+          std::vector<double> uplift_rate_in_m_year(vz.size());
+          for (size_t i = 0; i < vz.size(); ++i) {
+              uplift_rate_in_m_year[i] = vz[i] / year_in_seconds;
+          }
+
+          // 6. Create data arrays using grid shape
+          if (!elevation_initialized)
+          {
+            elevation = xt::adapt(h, shape);
+            elevation_initialized = true;
+            std::cout << "[DEBUG INIT] Initialized elevation from ASPECT solution.\n";
+          }
+          else
+          {
+            std::cout << "[DEBUG INIT] Reusing elevation from previous timestep.\n";
+          }
+          // auto elevation      = xt::adapt(h, shape);
+          auto elevation_old  = xt::adapt(h_old, shape);
+          auto uplift_rate    = xt::adapt(uplift_rate_in_m_year, shape);
+
+          //To fix the borders for box models
+          auto row_bounds = xt::view(uplift_rate, xt::keep(0, -1), xt::all());
+          row_bounds = 0.0;
+          auto col_bounds = xt::view(uplift_rate, xt::all(), xt::keep(0, -1));
+          col_bounds = 0.0;
```

Comment:
This is likely wrong. It tries to create views of `uplift_rate` as a 2-dimensional array, but `uplift_rate` is actually a 1-dimensional array with size = `n_grid_nodes` (i.e., the grid adapter's size).

I guess that the goal is to set fixed "boundary" conditions? This can be done in a grid-agnostic way by iterating over the flow graph's [base level nodes](https://fastscapelib.readthedocs.io/en/latest/api_cpp/flow.html#_CPPv4NK12fastscapelib10flow_graph11base_levelsEv), e.g.,

```cpp
for (auto& bi : flow_graph.base_levels())
{
   uplift_rate[bi] = 0.0;
}
```

(I haven't run and checked the code above)